analytics-integrations/moengage/src/main/java/com/segment/analytics/internal/integrations/MoEngageIntegration.java

```diff
@@ -0,0 +1,173 @@
+package com.segment.analytics.internal.integrations;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.Bundle;
+import android.util.Log;
+
+import com.moe.pushlibrary.MoEHelper;
+import com.moe.pushlibrary.utils.MoEHelperConstants;
+import com.segment.analytics.Analytics;
+import com.segment.analytics.AnalyticsContext;
+import com.segment.analytics.Traits;
+import com.segment.analytics.ValueMap;
+import com.segment.analytics.internal.AbstractIntegration;
+import com.segment.analytics.internal.model.payloads.IdentifyPayload;
+import com.segment.analytics.internal.model.payloads.TrackPayload;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.HashMap;
+import java.util.Set;
+
+import static com.segment.analytics.internal.Utils.hasPermission;
+
+/**
+ * MoEngage is an advanced mobile marketing and engagement tool which has a wide range of
+ * features. It helps user retention and increases churn.
+ *
+ * @see <a href="http://www.moengage.com/">MoEngage</a>
+ * @see <a href="https://segment.com/docs/integrations/moengage/">MoEngage Integration</a>
+ * @see <a href="http://docs.moengage.com/en/latest/android.html">MoEngage Android SDK</a>
+ */
+public class MoEngageIntegration extends AbstractIntegration<Void> {
+
+    private static final String KEY_MOENGAGE = "MoEngage";
+
+    private boolean setUserId = false;
+    MoEHelper mHelper;
+
+    private static final String ANONYMOUS_ID_KEY = "anonymousId";
+    private static final String EMAIL_KEY = "email";
+    private static final String USER_ID_KEY = "userId";
+    private static final String NAME_KEY = "name";
+    private static final String PHONE_KEY = "phone";
+    private static final String BIRTHDAY_KEY = "birthday";
+    private static final String FIRST_NAME_KEY = "firstName";
+    private static final String GENDER_KEY = "gender";
+    private static final String LAST_NAME_KEY = "lastName";
+    private static final String USER_ATTR_SEGMENT_AID = "USER_ATTRIBUTE_SEGMENT_ID";
+
+    private boolean hasLocationPermissions = false;
+
+    @Override
+    public void initialize(Analytics analytics, ValueMap settings) throws IllegalStateException {
+        Context context =  analytics.getApplication();
+        if (!hasPermission(context, "com.google.android.c2dm.permission.RECEIVE")) {
+            throw new IllegalStateException(
+                    "MoEngage requires com.google.android.c2dm.permission.RECEIVE permission");
+        }
+        if ( hasPermission(context, "android.permission.ACCESS_FINE_LOCATION")) {
+            hasLocationPermissions = true;
+        }
+        Analytics.LogLevel logLevel = analytics.getLogLevel();
+        if( logLevel == Analytics.LogLevel.VERBOSE || logLevel == Analytics.LogLevel.INFO ){
+            MoEHelper.APP_DEBUG = true;
+        }
+        mHelper = MoEHelper.getInstance(context);
```

Comment:
What is the point of this if we already have `mHelper = new MoEHelper(activity);` in the onActivityCreated method.